PWGMM/Mult/Tasks/dndeta-mft-pbpb.cxx

```diff
@@ -0,0 +1,1537 @@
+                          collision.posZ() - mcCollision.posZ(), c_rec);
+        } else {
+          QAregistry.fill(HIST("Events/ZvtxDiff"),
+                          collision.posZ() - mcCollision.posZ());
+        }
+
+        if (useZDiffCut) {
+          if (std::abs(collision.posZ() - mcCollision.posZ()) > maxZvtxDiff) {
+            continue;
+          }
+        }
+
+        if constexpr (C::template contains<aod::CentFT0Cs>()) {
+          registry.fill(HIST("Events/Centrality/NtrkZvtxGen"), nTrkRec,
+                        collision.posZ(), c_rec);
+        } else {
+          registry.fill(HIST("Events/NtrkZvtxGen"), nTrkRec, collision.posZ());
+        }
+      }
+    }
+
+    if constexpr (C::template contains<aod::CentFT0Cs>()) {
+      registry.fill(HIST("Events/Centrality/EvtEffGen"), 3., c_gen);
+    } else {
+      registry.fill(HIST("Events/EvtEffGen"), 3.);
+    }
+
+    auto perCollMCsample = mcSample->sliceByCached(
+      aod::mcparticle::mcCollisionId, mcCollision.globalIndex(), cache);
+    auto Nchrg = countPart(perCollMCsample);
+    if (moreThanOne > 1) {
+      if constexpr (C::template contains<aod::CentFT0Cs>()) {
+        QAregistry.fill(HIST("Events/Centrality/SplitMult"), Nchrg, c_gen);
+      } else {
+        QAregistry.fill(HIST("Events/SplitMult"), Nchrg);
+      }
+    }
+
+    auto zvtxMC = mcCollision.posZ();
+    auto nCharged = countPart(particles);
+    if constexpr (C::template contains<aod::CentFT0Cs>()) {
+      registry.fill(HIST("Events/Centrality/NtrkZvtxGen_t"), nCharged, zvtxMC,
+                    c_gen);
+    } else {
+      registry.fill(HIST("Events/NtrkZvtxGen_t"), nCharged, zvtxMC);
+    }
+
+    fillHist_MC<C::template contains<aod::CentFT0Cs>()>(particles, c_gen,
+                                                        zvtxMC, atLeastOne);
+
+    if (collisions.size() == 0) {
+      if constexpr (C::template contains<aod::CentFT0Cs>()) {
+        QAregistry.fill(HIST("Events/Centrality/NotFoundEventZvtx"),
+                        mcCollision.posZ(), c_gen);
+      } else {
+        QAregistry.fill(HIST("Events/NotFoundEventZvtx"), mcCollision.posZ());
+      }
+    }
+  }
+
+  /// @brief process fnc. to run on MC w/o centrality selection
+  void processMCInclusive(
+    filtMcGenColls::iterator const& mccollision,
+    soa::SmallGroups<soa::Join<filtColls, aod::McCollisionLabels>> const& collisions,
+    filtParticles const& particles, filtMcMftTracks const& tracks)
+  {
+    processMC<filtMcGenColls, filtColls>(mccollision, collisions, particles,
+                                         tracks);
+  }
+
+  PROCESS_SWITCH(PseudorapidityDensityMFT, processMCInclusive,
+                 "Count MC particles", false);
+
+  /// @brief process fnc. to run on MC w FT0C centrality selection
+  void processMCCent(
+    filtMcGenColls::iterator const& mccollision,
+    soa::SmallGroups<soa::Join<filtCollsCent, aod::McCollisionLabels>> const& collisions,
```

Comment:
Note that having `Filtered` not as an outermost template (although inside `SmallGroups`) makes it ineffective. If you want the filters to be applied to collisions here, you need
`soa::SmallGroups<soa::Filtered<soa::Join<aod::Collisions, aod::EvSels, aod::McCollisionsLabels>>>>`

Reply:
Okay, I will fix that in another commit later on. Thanks for taking a careful look.